### PR TITLE
Update Synthetix Debt Pool EA to reflect SIP-165 changes

### PR DESCRIPTION
**Pull request by cds95**
## Description

SIP-165 has been updated to require two oracles.  The first oracle will aggregate the total debt issued across multiple chains and the second calculates the debt ratio across all chains.  This PR adds support for this by creating two endpoints in the Synthetix Debt Pool EA named `debt` and `debt-ratio`.

https://sips-git-165-oracle-synthetixio.vercel.app/sips/sip-165/

## Changes

- Add two endpoints to Synthetix Debt Pool EA named `debt` and `debt-ratio`.
- Update unit and integration tests
- Refactor EA to use the SynthetixIO TypeScript library

## Steps to Test

1. Set env vars as instructed in the ReadME and run the EA
2. Verify successful response from the EA in the following scenarios
  - Get debt from a single chain
  - Get debt across multiple chains
  - Get debt ratio from a single chain
  - Get debt ratio across multiple chains

## Quality Assurance

- [x] Ran `yarn changeset` if adapter source code was changed
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/sources/synthetix-debt-pool/src/commons.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    data: {
+      result: debt.toString(),
+    },
+  }
+  return Requester.success(jobRunID, result, config.verbose)
+}
+
+const validateChainSources = (jobRunID: string, chainSources: string[]) => {
+  for (const source of chainSources) {
+    const supportedChains = Object.values(SupportedChains) as string[]
```
Comment (cds95) [inline packages/sources/synthetix-debt-pool/src/commons.ts:43]: declare this outside of the for loop

File: packages/sources/synthetix-debt-pool/src/commons.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+): SynthetixJS => {
+  const { rpcURL, networkId } = config.chains[network]
+  if (!rpcURL) {
+    throw new AdapterError({
+      jobRunID,
+      message: `RPC URL not set for chain: ${network}`,
+    })
+  }
+  const provider = new ethers.providers.JsonRpcProvider(rpcURL)
+  return synthetix({ provider, networkId: networkId })
```
Comment (cds95) [inline packages/sources/synthetix-debt-pool/src/commons.ts:66]: can just be `networkId`

File: .changeset/lazy-crews-mate.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@chainlink/synthetix-debt-pool-adapter': major
+---
+
+Update Synthetix Debt Pool EA to reflect changes to SIP-165. This separates the EA to 2 endpoints. One is to calculate the Debt Ratio and the other is to calcualte the total debt issued. The chain names that are passed in the request body have also been updated to match those in the SynthetixIO JavaScript library.
```
Comment (austinborn) [inline .changeset/lazy-crews-mate.md:5]:
```suggestion
Update Synthetix Debt Pool EA to reflect changes to SIP-165. This separates the EA to 2 endpoints. One is to calculate the Debt Ratio and the other is to calculate the total debt issued. The chain names that are passed in the request body have also been updated to match those in the SynthetixIO JavaScript library.
```

File: packages/sources/synthetix-debt-pool/src/commons.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import { AdapterError } from '@chainlink/ea-bootstrap'
+import { SynthetixJS, synthetix } from '@synthetixio/contracts-interface'
+
+const inputParameters: InputParameters = {
+  chainSources: {
+    required: false,
+    description: `Array of chains to pull debt from. Options for array elements are "mainnet", "mainnet-ovm", "kovan", "kovan-ovm"`,
+    type: 'array',
+  },
+}
```
Comment (austinborn) [inline packages/sources/synthetix-debt-pool/src/commons.ts:14]: In order for this to work with the doc generator scripts, we need to export this from each endpoint. Would it be ok to export this from each endpoint somehow? Either copying the object in each, or maybe import and re-export?

**Review by austinborn — CHANGES_REQUESTED** [review]

File: packages/sources/synthetix-debt-pool/src/commons.ts (hunk 4 above)
Comment (cds95) [reply packages/sources/synthetix-debt-pool/src/commons.ts:14]: sure I think we can do that

**Review by austinborn — DISMISSED** [review]
Awesome, thank you

**Review by boxhock — APPROVED** [review]

**Pull request merged**